upgrade/class/Xoops/Upgrade/UpgradeControl.php

```
@@ -0,0 +1,325 @@
+            : false;
+
+        if (false !== $languageRoot && false !== $candidate && str_starts_with($candidate, $languageRoot . DIRECTORY_SEPARATOR)) {
+            include_once $candidate;
+        } elseif (false !== $languageRoot && false !== $fallback && str_starts_with($fallback, $languageRoot . DIRECTORY_SEPARATOR)) {
+            include_once $fallback;
+        }
+
+        if (null !== $supports) {
+            $this->supportSites = array_merge($this->supportSites, $supports);
+        }
+    }
```

PHPMD (2.15.0)

[warning] The method loadLanguage() has a Cyclomatic Complexity of 10. The configured cyclomatic complexity threshold is 10. (undefined)

(CyclomaticComplexity)